atl-checker/src/algorithms/game_strategy/partial.rs

```diff
@@ -0,0 +1,275 @@
+                    // negation edge to an DespiteUntil. Let's visit that one instead.
+                    let edges = graph.annotated_succ(vertex);
+                    if let Some(AnnotatedEdge::Negation(edge)) = edges.get(0) {
+                        println!("enf inv {:?}", assignments.get(&vertex));
+                        compute_partial_strategy_rec(
+                            graph,
+                            &edge.target,
+                            assignments,
+                            move_to_pick,
+                        );
+                    }
+                }
+                Phi::EnforceEventually { .. } | Phi::EnforceUntil { .. } => {
+                    let mut edges = graph.annotated_succ(vertex);
+                    let mut edges_drain = edges.drain(..);
+
+                    // Check if subformula is true in this state. We can do this by checking if the
+                    // target of the first edge is true
+                    if let Some(AnnotatedEdge::Hyper(edge)) = edges_drain.next() {
+                        let all_targets_true = edge.targets.iter().all(|(t, _)| {
+                            matches!(assignments.get(t), Some(VertexAssignment::True))
+                        });
+
+                        if all_targets_true {
+                            // No need for more moves
+                            return;
+                        }
+                    }
+
+                    // Find the first hyper-edge where all targets are true
+                    for edge in edges_drain {
+                        if let AnnotatedEdge::Hyper(edge) = edge {
+                            let all_targets_true = edge.targets.iter().all(|(t, _)| {
+                                matches!(assignments.get(t), Some(VertexAssignment::True))
+                            });
+
+                            if all_targets_true {
+                                // We have found the partial move that will guarantee the property
+                                move_to_pick.insert(*state, edge.annotation.unwrap());
+
+                                // Recurse to find the moves of the next states
+                                for (target, _) in edge.targets {
+                                    compute_partial_strategy_rec(
+                                        graph,
+                                        &target,
+                                        assignments,
+                                        move_to_pick,
+                                    );
+                                }
+                                return;
+                            }
+                        }
+                    }
+                }
+                Phi::DespiteNext { .. } => {
+                    let edges = graph.annotated_succ(vertex);
+                    // There is only one hyper-edge
+                    for edge in edges {
```

Comment:
This i have no opinion on, but could also be reduced to
```suggestion
                    // There is only one hyper-edge
                    for edge in  graph.annotated_succ(vertex) {
```